src/payment/instrument/instrument-selector.ts

```diff
@@ -19,7 +22,25 @@ export type InstrumentSelectorFactory = (state: InstrumentState) => InstrumentSe
 export function createInstrumentSelectorFactory(): InstrumentSelectorFactory {
     const getInstruments = createSelector(
         (state: InstrumentState) => state.data,
-        instruments => () => instruments
+        instruments => (paymentMethod?: PaymentMethod) => {
+            if (!instruments) {
+                return undefined;
+            }
+
+            if (!paymentMethod) {
+                const supportedCardInstruments = filter(supportedInstruments, { method: 'card' });
+
+                return flatMap(supportedCardInstruments, cardInstrument => filter(instruments, cardInstrument));
```

Comment:
Is the order of `instruments` important? i.e.: Right now, it seems the order of `Instrument` is based on `supportedInstruments` and then `instruments`. If it's not important, then what you have here is good :)

Reply:
We only show one instrument type at a given time, so as long as we retain the order for every type of instrument it should be ok.